LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs

```diff
@@ -1047,6 +1035,918 @@ public static void CopyMatToBMOorMIC(PackageEditorWindow pew)
+                        equal = equal && (unparsed[idBase + i] == oldArr[i]);
+                    }
+
+                    // Replace the hash
+                    if (equal)
+                    {
+                        for (int i = 0; i < 4; i++)
+                        {
+                            unparsed[idBase + i] = newArr[i];
+                        }
+
+                        hirc.unparsed = unparsed;
+                    }
+                }
+            }
+
+            wwiseBankEntry.WriteBinary(wwiseBank);
+
+            if (!called)
+            {
+                MessageBox.Show("WwiseBank's ID updated successfully.", "Success", MessageBoxButton.OK);
+            }
+        }
+
+        /// <summary>
+        /// Rename a bioConversation's FXAs, WwiseStreams, and VOElements names.
+        /// Not useful as a standalone experiment, since other semi-unrelated elements need to be renamed too.
+        /// </summary>
+        /// <param name="pew">Current PE window.</param>
+        /// <param name="bioConversation">Conversation the elements belong to.</param>
+        /// <param name="oldName">Old name to replace in the elements.</param>
+        /// <param name="newName">New name to replace in the elements.</param>
+        private static void RenameFXAsAndRelated(PackageEditorWindow pew, ExportEntry bioConversation, string oldName, string newName)
+        {
+            if (pew.Pcc == null || pew.SelectedItem?.Entry == null || (pew.Pcc.Game is MEGame.ME1))
+            {
+                return;
+            }
+
+            List<ObjectProperty> fxas = new();
+            ArrayProperty<ObjectProperty> maleFXAs = bioConversation.GetProperty<ArrayProperty<ObjectProperty>>("m_aMaleFaceSets");
+            if (maleFXAs != null) { fxas.AddRange(maleFXAs); }
+            ArrayProperty<ObjectProperty> femaleFXAs = bioConversation.GetProperty<ArrayProperty<ObjectProperty>>("m_aFemaleFaceSets");
+            if (femaleFXAs != null) { fxas.AddRange(femaleFXAs); }
+            ObjectProperty nonSpkrFxa = bioConversation.GetProperty<ObjectProperty>("m_pNonSpeakerFaceFXSet");
+            if (nonSpkrFxa != null) { fxas.Add(nonSpkrFxa); }
+
+            foreach (ObjectProperty fxa in fxas)
+            {
+                if (fxa.Value == 0) { continue; }
+
+                ExportEntry fxaExport = pew.Pcc.GetUExport(fxa.Value);
+
+                string oldFxaFullName = fxaExport.ObjectName; // May contain _M, _F, or _NonSpkr
+                string oldFxaName = oldFxaFullName; // Full name minus _M/_F, or including _NonSpkr
+
+                if (oldFxaFullName[^2..].ToLower() is "_m" or "_f")
+                {
+                    oldFxaName = oldFxaFullName.Remove(oldFxaFullName.Length - 2);
+                }
+                else
+                {
+                    // Most likely a NonSpkr, in which case we'll use the full name
+                    if (oldFxaFullName.Length > 8 && oldFxaFullName[^8..].ToLower() is "_nonspkr")
```

Comment:
```suggestion
                    if (oldFxaFullName.EndsWith("_nonspkr", StringComparison.OrdinalIgnoreCase))
```

Reply:
This is what I get for not geting acquainted with the available methods, hahaha

Reply:
I'm actually gonna have to go and change something at the animations experiment with this knowledge